// @feynman/protocol/dist/index.js
/**
 * @feynman/protocol — wire types for the `/v1/*` HTTP+SSE surface defined in
 * ARCHITECTURE.md §5. Types only; this package emits no runtime code.
 *
 * The plugin (`feynman-obsidian`) and the server (`@feynman/server`) both import
 * from here so the wire stays in lockstep. Pi internals MUST NOT leak through
 * this file — every type here corresponds to a public `/v1/*` shape.
 */
// ---------------------------------------------------------------------------
// 5.1 Endpoint paths
// ---------------------------------------------------------------------------
/**
 * Canonical URL templates for every endpoint in ARCHITECTURE.md §5.1.
 * Functions take path params and return the path; static endpoints are plain
 * strings. No base URL — callers prepend it.
 */
export const PATHS = {
    health: "/v1/health",
    manifest: "/v1/manifest",
    run: "/v1/run",
    runState: (runId) => `/v1/runs/${runId}`,
    runEvents: (runId) => `/v1/runs/${runId}/events`,
    runInput: (runId) => `/v1/runs/${runId}/input`,
    runCancel: (runId) => `/v1/runs/${runId}/cancel`,
    runArtifact: (runId, path) => `/v1/runs/${runId}/artifacts/${path}`,
    licenseActivate: "/v1/license/activate",
};
